apps/web/utils/discord-markdown.ts

```diff
@@ -3,77 +3,158 @@ import { load } from 'cheerio'
+
+  return {
+    posts: postsFiltered,
+    users: usersFiltered,
+    channels: channelsFiltered
+  }
+}
+
+const internalLink = (content: string, posts: PostCache[]) => {
+  // Replace internal links
+  content = content.replace(channelLinkRegex, (match, guildId, channelId, _, messageId) => {
+    const post = posts.find((p) => p.snowflakeId === channelId)
     if (!post) return match
-    return `${getCanonicalPostUrl(post.snowflakeId)}${message ? `#message-${message}` : ''}`
+    return `${getCanonicalPostUrl(post.snowflakeId)}${messageId ? `#message-${messageId}` : ''}`
   })
+  return content
+}
 
-  // The library doesn't allow async callbacks, so we have to do this in two steps
-  toHTML(content, {
-    discordOnly: true,
-    discordCallback: {
-      user(node) {
-        memberIds.add(node.id)
-        return ''
-      },
-      channel(node) {
-        channelIds.add(node.id)
-        return ''
-      },
-    },
+export const parseDiscordMessageBasic = async (content: string) => {
+  // Get mentions
+  const { users, channels, posts } = await fetchMentions(content)
+
+  // Replace user mentions
+  content = content.replace(userMention, (match, userId) => {
+    const member = users.find((u) => u.snowflakeId === userId)
+    const userName = sanitizeText(member?.username ?? 'Unknown User')
+    return `@${userName}`
+  })
+
+  // Replace channel mentions
+  content = content.replace(channelMention, (match, channelId) => {
+    const channel = channels.find((c) => c.snowflakeId === channelId)
+    let channelName = channel && sanitizeText(channel.name)
+
+    if (!channelName) {
+      const post = posts.find((p) => p.snowflakeId === channelId)
+      channelName = post ? sanitizeText(post.title) : 'Unknown Channel'
+    }
+    return `#${channelName}`
   })
 
-  const users =
-    memberIds.size > 0
-      ? await db
-        .selectFrom('users')
-        .select(['snowflakeId', 'username'])
-        .where('snowflakeId', 'in', Array.from(memberIds))
-        .execute()
-      : []
-
-  const channels =
-    channelIds.size > 0
-      ? await db
-        .selectFrom('channels')
-        .select(['snowflakeId', 'name'])
-        .where('snowflakeId', 'in', Array.from(channelIds))
-        .execute()
-      : []
-
-  // Parse the content again, this time replacing the nodes and the rest of the stuff
+  // Replace internal links
+  content = internalLink(content, posts)
+
+  return content
+}
+
+export const parseDiscordMessage = async (content: string) => {
+  // Get mentions
+  const { users, channels, posts } = await fetchMentions(content)
+
+  // Replace internal links
+  content = internalLink(content, posts)
+
+  // Parse the content
   const html = toHTML(content, {
     discordCallback: {
       user: (node) => {
         const user = users.find((u) => u.snowflakeId === node.id)
-        const userName = sanitizeText(user?.username ?? 'Unknown User')
+        if (!user) return `<i>@Unknown User</i>`
```

Comment:
What is the reason to wrap only unknown users with `<i>`?

Reply:
On discord, unknown menuones are italics, so I wanted to continue that.